src/cluster.c

```diff
@@ -3651,6 +3652,19 @@ void clusterCron(void) {
         }
     }
 
+    /* Iterate nodes to remove free space in the send buffer
+    * to reclaim memory. */
+    di = dictGetSafeIterator(server.cluster->nodes);
```

Comment:
It seems there are 2 other places in `clusterCron()` where we perform various maintenance on the `clusterNode`s. Maybe it's time to refactor this we have a single `clusterNodeCron()` where we do everything instead of adding yet another iteration? @yossigo? @madolson?

Reply:
+1, I like this idea.

Reply:
I can look at refactoring all the for loops into a single function. I assume we want something like `clientsCron`?

Reply:
Yes. That would be great.